core/frontend/src/Viewport.ts

```diff
@@ -1231,6 +1231,7 @@ export abstract class Viewport implements IDisposable, TileUser {
     removals.push(settings.contextRealityModels.onPlanarClipMaskChanged.addListener(displayStyleChanged));
     removals.push(settings.contextRealityModels.onAppearanceOverridesChanged.addListener(displayStyleChanged));
     removals.push(settings.contextRealityModels.onDisplaySettingsChanged.addListener(displayStyleChanged));
+    removals.push(settings.contextRealityModels.onInvisibleChanged.addListener(displayStyleChanged));
```

Comment:
You need to `invalidateController` (which will invalidate the render plan), because toggling visibility can affect the viewed volume. (re: your comment about frustum culling).

Reply:
ok, do I also need a changeFlag ?

Reply:
If you mean `this._changeFlags.setDisplayStyle()`, yes.